[docs] Add an integration guideline page for In-App purchases (#26425)
expo/expo: docs/pages/guides/in-app-purchases.mdx

---
title: Use In-app purchase
description: Learn about how to use in-app purchases in your Expo app.
hideTOC: true
---

import { BoxLink } from '~/ui/components/BoxLink';
import { CODE } from '~/ui/components/Text';
import { BookOpen02Icon, GithubIcon } from '@expo/styleguide-icons';

In-app purchases (IAP) are transactions within a mobile or desktop application where users can buy digital goods or additional features. This guide provides a list of popular libraries and tutorials for implementing IAP in your Expo app.

> > In-app purchase libraries require configuring custom native code. Native code is not configurable when using Expo Go. Instead, create a [development build](/develop/development-builds/introduction/), which will allow you to use a native library in your project.

Author: ```suggestion
> In-app purchase libraries require configuring custom native code. Native code is not configurable when using Expo Go. Instead, create a [development build](/develop/development-builds/introduction/), which allows using a native library in your project.
```

## Tutorial

<BoxLink
  title="Expo In-App Purchase Tutorial"
  description={
    <>
      The getting started guide for in-app purchases and subscriptions with{' '}
      <CODE>react-native-purchases</CODE> library and RevenueCat.
    </>
  }
  href="https://www.revenuecat.com/blog/engineering/expo-in-app-purchase-tutorial/"
  Icon={BookOpen02Icon}
/>

## Libraries

> 

Reviewer: ```suggestion

These libraries provide robust support for in-app purchase functionality, as well as out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/). 
```
Some kind of intro as to why we're recommending these (they have config plugins + Expo-specific docs)

<BoxLink
  title={
    <>
      <CODE>react-native-purchases</CODE>
    </>
  }

>   description="An open-source framework that provides a wrapper around Google Play Billing, StoreKit and RevenueCat. This library is a client for RevenueCat's subscription and purchase tracking system."

Reviewer: ```suggestion
  description="An open-source framework that provides a wrapper around Google Play Billing and StoreKit API's, as well as integration with RevenueCat services supporting in-app purchases. `react-native-purchases` enables product management, analytics, and simplified workflows for in-app purchase requirements that may extend beyond your client code, such as validating purchases on your application backend."
```
maybe not the best wording, but wanted to emphasize the receipt validation angle, that's got a lot of appeal even if you don't like the other stuff.

  href="https://github.com/RevenueCat/react-native-purchases"
  Icon={GithubIcon}
/>

<BoxLink
  title={
    <>

>       <CODE>react-native-iap</CODE>

Author: @keith-kurak, is it worth mentioning this library? In their [docs] or repo I do not see anything mentioned about Expo.

Also, are there any other libraries or resources we should mention here?

    </>
  }

>   description="A React Native library to access In-app purchases capabilities in your Android and iOS apps."

Reviewer: ```suggestion
  description="A React Native library providing an interface to the client-side native in-app purchase API's built into Android and iOS."
```
Trying to emphasize more that it's "just the wrapper".

---
title: Use In-app purchase
description: Learn about how to use in-app purchases in your Expo app.
hideTOC: true
---

import { BoxLink } from '~/ui/components/BoxLink';
import { CODE } from '~/ui/components/Text';
import { BookOpen02Icon, GithubIcon } from '@expo/styleguide-icons';

In-app purchases (IAP) are transactions within a mobile or desktop application where users can buy digital goods or additional features. This guide provides a list of popular libraries and tutorials for implementing IAP in your Expo app.

> In-app purchase libraries require configuring custom native code. Native code is not configurable when using Expo Go. Instead, create a [development build](/develop/development-builds/introduction/), which allows using a native library in your project.

## Tutorial

<BoxLink
  title="Expo In-App Purchase Tutorial"
  description={
    <>
      The getting started guide for in-app purchases and subscriptions with{' '}
      <CODE>react-native-purchases</CODE> library and RevenueCat.
    </>
  }
  href="https://www.revenuecat.com/blog/engineering/expo-in-app-purchase-tutorial/"
  Icon={BookOpen02Icon}
/>

## Libraries

> These libraries provide robust support for in-app purchase functionality, as well as out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/). 

Author: ```suggestion
These libraries provide robust support for in-app purchase functionality and out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/).

```

<BoxLink
  title={
    <>
      <CODE>react-native-purchases</CODE>
    </>
  }

>   description="An open-source framework that provides a wrapper around Google Play Billing and StoreKit API's, as well as integration with RevenueCat services supporting in-app purchases. `react-native-purchases` enables product management, analytics, and simplified workflows for in-app purchase requirements that may extend beyond your client code, such as validating purchases on your application backend."
>   href="https://github.com/RevenueCat/react-native-purchases"
>   Icon={GithubIcon}
> />

Author: ```suggestion
  description={<><CODE>react-native-purchases</CODE> is an open-source framework that provides a wrapper around Google Play Billing and StoreKit APIs, as well as integration with RevenueCat services supporting in-app purchases. It enables product management, analytics, and simplified workflows for in-app purchase requirements that may extend beyond your client code, such as validating purchases on your app's backend.</>}
  href="https://github.com/RevenueCat/react-native-purchases"
  Icon={GithubIcon}
/>
```

---
title: Use In-app purchase
description: Learn about how to use in-app purchases in your Expo app.
hideTOC: true
---

import { BoxLink } from '~/ui/components/BoxLink';
import { CODE } from '~/ui/components/Text';
import { BookOpen02Icon, GithubIcon } from '@expo/styleguide-icons';

In-app purchases (IAP) are transactions within a mobile or desktop application where users can buy digital goods or additional features. This guide provides a list of popular libraries and tutorials for implementing IAP in your Expo app.

> In-app purchase libraries require configuring custom native code. Native code is not configurable when using Expo Go. Instead, create a [development build](/develop/development-builds/introduction/), which allows using a native library in your project.

## Tutorial

<BoxLink
  title="Expo In-App Purchase Tutorial"
  description={
    <>
      The getting started guide for in-app purchases and subscriptions with{' '}
      <CODE>react-native-purchases</CODE> library and RevenueCat.
    </>
  }
  href="https://www.revenuecat.com/blog/engineering/expo-in-app-purchase-tutorial/"
  Icon={BookOpen02Icon}
/>

## Libraries

> These libraries provide robust support for in-app purchase functionality and out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/).

Author: ```suggestion
The following libraries provide robust support for in-app purchase functionality and out-of-the-box compatibility with Expo apps using [CNG](workflow/continuous-native-generation/) and provide [Config Plugins](/config-plugins/introduction/) for seamless integration in your app.
```
